packages/experiments/src/components/SelectedItemsList/ControlledSelectedItemsList.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+
+import { IControlledSelectedItemsList, IControlledSelectedItemListProps, BaseSelectedItem } from './SelectedItemsList.types';
+import { copyToClipboard } from '../../utilities/copyToClipboard';
+import { useSelectionIndices } from '../../utilities/useSelectionIndices';
+
+/**
+ * An externally-managed selected item list.
+ */
+export const ControlledSelectedItemsList = <TItem extends BaseSelectedItem>(props: IControlledSelectedItemListProps<TItem>) => {
+  const selection = React.useMemo(() => props.selection, [props.selection]);
+  const selectedIndices = useSelectionIndices(selection);
+  // Synchronize the selection against the items in the selection
+  React.useEffect(() => {
+    selection &&
+      selection.setItems(
+        props.selectedItems,
```

Comment:
I'm of mixed opinion around this.

I originally put this here so someone consuming the component as controlled wouldn't have to manage the selection themselves. Within fabric as it is now, there's a standard where even when a component is 'controlled', smart objects like Selection still have their lifecycles managed by inner components.

I guess where it should live depends on whether or not "Selection" counts as part of the data model of a component.

Reply:
If it does, I see the argument for pulling this out of the Controlled component.

However, unless we wanted to force users to manage the selection object's lifecycle, we would have to present a different component to the user (e.g. one that only manages selection state and doesn't manipulate the actual data).

Reply:
I feel like the view shouldn't have an opinion of how the selection is given to it, there are other ways to manage selection that the consuming component might want to use, at least it seems like it might. The consumer doesn't necessarily need to use the Selection class right?